### PR TITLE
logs is done

### PR DESCRIPTION
**Pull request by vladkuzya**

### CODE REVIEW

**Review by IrynaDim — COMMENTED** [review]
At QA, we were told that we could also add an info logger after transaction.commit or transaction.update in our DAOs. To show that our entity was created or updated.

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 .idea/*
 *.iml
-target/*
\ No newline at end of file
+target/*
+logs/*
```
Comment (KseniiaMakarova) [inline .gitignore]: empty line

File: src/main/resources/log4j.properties [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Root logger options
+log4j.rootLogger=DEBUG, CONSOLE, FILE
+
+# CONSOLE appender customisation
+log4j.appender.CONSOLE=org.apache.log4j.ConsoleAppender
+log4j.appender.CONSOLE.threshold=DEBUG
+log4j.appender.CONSOLE.layout=org.apache.log4j.PatternLayout
```
Comment (KseniiaMakarova) [inline src/main/resources/log4j.properties]: let's add log4j.appender.stdout.Target

File: src/main/java/mate/academy/Main.java [hunk 3]
```diff
@@ -38,39 +40,39 @@ public static void main(String[] args) throws AuthenticationException {
...
 
         shoppingCartService.addSession(movieSession, bob);
         ShoppingCart shoppingCartBob = shoppingCartService.getByUser(bob);
-        System.out.println(shoppingCartBob);
+        logger.info(shoppingCartBob);
         shoppingCartService.clear(shoppingCartBob);
-        System.out.println(shoppingCartBob);
+        logger.info(shoppingCartBob);
 
         User alisa = authenticationService.login("alisa@gmail.com", "1111");
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/Main.java]: let's wrap call to `login()` method in try-catch and add log.warn with the informative message in `catch` block

File: src/main/java/mate/academy/Main.java [hunk 4]
```diff
@@ -38,39 +40,39 @@ public static void main(String[] args) throws AuthenticationException {
...
-        System.out.println(userService.findByEmail("alica@gmail.com"));
+        logger.info(userService.findByEmail("alica@gmail.com"));
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/Main.java]: it will be nice to add some informative text along (reconsider all your logs)

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]
I agree with Irina - let's also add logs to the DaoImpl classes

File: src/main/java/mate/academy/dao/impl/AbstractDao.java [hunk 5]
```diff
@@ -33,8 +39,8 @@ public T add(T entity, Class<T> clazz) {
         try (Session session = HibernateUtil.getSessionFactory().openSession()) {
             return session.createQuery("from " + clazz.getSimpleName(), clazz).getResultList();
         } catch (Exception e) {
-            throw new DataProcessingException("Error retrieving all "
-                    + clazz.getSimpleName(), e);
+            logger.error("Cant fetch all {} from DB", clazz.getName(), e);
+            return Collections.emptyList();
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/dao/impl/AbstractDao.java]: exceptions are logged by default - we should not log them
let's return to the original version here

File: src/main/java/mate/academy/dao/impl/MovieSessionDaoImpl.java [hunk 6]
```diff
@@ -24,9 +28,11 @@
             query.setParameter("endDate", date.atTime(LocalTime.MAX));
             return query.getResultList();
         } catch (Exception e) {
-            throw new DataProcessingException("Can't find available sessions at this date"
-                    + "for movie with id" + movieId, e);
+            logger.error("Cant fetch MovieSessions for movieId={}, date={} from DB",
+                    movieId, date, e);
+            return Collections.emptyList();
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/dao/impl/MovieSessionDaoImpl.java]: same everywhere please

File: src/main/java/mate/academy/dao/impl/OrderDaoImpl.java [hunk 7]
```diff
@@ -23,6 +28,9 @@ public Order add(Order order) {
                     + "WHERE o.user = :user", Order.class)
                     .setParameter("user", user)
                     .getResultList();
+        } catch (Exception e) {
+            logger.error("Can't fetch User {} from DB", user, e);
+            return Collections.emptyList();
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/dao/impl/OrderDaoImpl.java]: there is already `catch` in super.add()

File: src/main/resources/log4j.properties [hunk 8]
```diff
@@ -1,17 +1,18 @@
...
-log4j.appender.FILE.MaxBackupIndex=2
-log4j.appender.FILE.layout=org.apache.log4j.PatternLayout
-log4j.appender.FILE.layout.ConversionPattern=[ %-5p] - %c:%L - %m%n
+log4j.rootLogger=WARN, file, stdout
+log4j.logger.org.hibernate=INFO
+log4j.logger.org.hibernate.SQL=INFO
+# Direct log messages to a log file
+log4j.appender.file=org.apache.log4j.RollingFileAppender
+log4j.appender.file.threshold=INFO
+log4j.appender.file.File=./target/logging/logging.log
```
Comment (KseniiaMakarova) [inline src/main/resources/log4j.properties]: this should point to the `logs` folder that you added to `gitignore`
`log4j.appender.FILE.File=./logs/logging.log` before it looked better

File: pom.xml [hunk 9]
```diff
@@ -40,9 +40,14 @@
...
-            <version>1.2.17</version>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-api</artifactId>
+            <version>1.7.30</version>
+        </dependency>
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-log4j12</artifactId>
+            <version>1.7.30</version>
         </dependency>
```
Comment (KseniiaMakarova) [inline pom.xml]: does it work well with `log4j.properties`? if yes, it is ok

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 9 above)
Comment (vladkuzya) [reply pom.xml]: Yes, it works)

File: src/main/java/mate/academy/dao/impl/UserDaoImpl.java [hunk 10]
```diff
@@ -8,9 +8,13 @@
 import mate.academy.util.HibernateUtil;
 import org.hibernate.Session;
 import org.hibernate.query.Query;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 @Dao
 public class UserDaoImpl extends AbstractDao<User> implements UserDao {
+    private static final Logger logger = LoggerFactory.getLogger(UserDaoImpl.class);
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/dao/impl/UserDaoImpl.java]: remove this

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**